### ISSUE TITLE
Nested class page not shown in hierarchy

### ISSUE THREAD

**Issue by rnett** [Kotlin/dokka#1879]
**Describe the bug**
I have a class like:
```kotlin
sealed class KrosstalkResult{
    data class Success(): KrosstalkResult()
    data class HttpError(): KrosstalkResult()
}
```

Nested classes are shown properly on the `KrosstalkResult` page.  However, navigating to one of the nested class's pages results in this:
![image](https://user-images.githubusercontent.com/32281897/115787679-70923d80-a377-11eb-9403-036fb2e3357b.png)

The page is correct, but the navigation sidebar is collapsed instead of showing the location of the class.

**Expected behaviour**
Nested class pages are present in the navigation sidebar, and their location is shown in the sidebar when they are opened.

**Screenshots**
If applicable, add screenshots to help explain your problem

**To Reproduce**
https://github.com/rnett/krosstalk/blob/5e5a8cc4ed0120d25e0e0dd084796c583aa665aa/build.gradle.kts

**Dokka configuration**
https://github.com/rnett/krosstalk/blob/5e5a8cc4ed0120d25e0e0dd084796c583aa665aa/build.gradle.kts

**Installation**
- Operating system: Windows
- Build tool: Gradle v6.8.3
- Dokka version: 1.4.32

**Comment by DanielRibeiroMaciel-TomTom** [comment]
We also experience the same problem.

**Comment by IngoHaumann-TomTom** [comment]
This issue is still present in version 1.7.10.

**Comment by IgnatBeresnev** [comment]
Indeed. Just for the reference: https://rnett.github.io/krosstalk/release/core/krosstalk/com.rnett.krosstalk.result/-krosstalk-result/-http-error/index.html

**Comment by IgnatBeresnev** [comment]
Looks like this has been implemented in #2597, released in 1.7.20. I tried to write a similar reproducer, works fine. Please let me know if I understood it wrong.

![2023-03-08_21-46-13](https://user-images.githubusercontent.com/22685910/223846041-471880be-d8d6-4d1a-b66b-3bace38d8cf9.png)
